### PR TITLE
Basic circuit vizualization

### PR DESCRIPTION
**Pull request by GiggleLiu**
```julia
using Yao
import YaoExtensions
qft = YaoExtensions.qft_circuit(4)

using YaoPlots, Compose, Cairo
CircuitStyles.setfontfamily("ubuntu")
CircuitStyles.setparamtextsize(8pt)

qft |> vizcircuit(scale=0.7) |> PNG("qft.png")
```

qft

```julia
c2 = YaoExtensions.variational_circuit(4, 2)
c2 |> vizcircuit(scale=0.7) |>  PNG("variational.png")
```
variational

### CODE REVIEW

**Comment by GiggleLiu** [comment]
Julia 1.0 test breaks. Shall we just drop the v1.0 support?

File: src/vizcircuit.jl [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+module CircuitStyles
+	using Compose
+    r = 0.2
+	lw = 1pt
+	textsize = 16pt
+	paramtextsize = 12pt
+	fontfamily = "Helvetica Neue"
+    G() = compose(context(), rectangle(-r, -r, 2r, 2r), fill("white"), stroke("black"), linewidth(lw))
+    C() = compose(context(), circle(0.0, 0.0, r/3), fill("black"))
+    X() = compose(context(), xgon(0.0, 0.0, r, 4), fill("black"))
```
Comment (ChenZhao44) [inline src/vizcircuit.jl]: Should we add the `oplus` for the conventional CNOT notation?

**Review by ChenZhao44 — COMMENTED** [review]
Only a small suggestion

**Comment by GiggleLiu** [comment]
variational

File: src/vizcircuit.jl [hunk 2]
```diff
@@ -0,0 +1,159 @@
+using Viznet: canvas
+using YaoBlocks
+
+export CircuitStyles, CircuitGrid, circuit_canvas, vizcircuit
+
+module CircuitStyles
+	using Compose
+    r = 0.2
```
Comment (Roger-luo) [inline src/vizcircuit.jl]: why these are not `const Ref`? the canonical way of implementing global tweakable options is to use e.g

```jl
const r = Ref{Float64}(0.2)
```

**Comment by Roger-luo** [comment]
how do you format the floating points in phase? I think it should be printed using `@printf/@sprintf` or they can be too long.

**Comment by Roger-luo** [comment]
also can we just use `plot`? the function can be dispatched directly on `blocks` etc. I think this package only needs one function `plot` for simple cases.

**Comment by GiggleLiu** [comment]
> how do you format the floating points in phase? I think it should be printed using `@printf/@sprintf` or they can be too long.

use `round(x, digits=2)`

**Comment by GiggleLiu** [comment]
See https://github.com/QuantumBFS/YaoPlots.jl/tree/circuit/examples for more circuit demos.

**Pull request merged**